torch/__init__.py

```diff
@@ -1752,6 +1752,90 @@ def registerOp(cls, op_key, full_schema, op_impl, dispatch_key):
+    constraints: Optional[List["torch._dynamo.eval_frame.Constraint"]] = None,
+) -> "torch._export.exported_program.ExportedProgram":
+    """
+    torch.export() is a one-shot process for capturing a computation graph from
+    a PyTorch program Ahead-of-Time (AOT).
+
+    This function traces either an nn.Module's forward function or a callable
+    containing PyTorch operations and produces an ExportedProgram. The
+    ExportedProgram includes PyTorch operations that perform computations
+    equivalent to those in the given nn.Module or callable.
+
+    In specific terms, torch.export() traces a function `f` by executing it
+    with the provided `args` and `kwargs`. It records the PyTorch operations
+    invoked during execution to produce the ExportedProgram.
+
+    Acceptable types of inputs for `args` and `kwargs` include:
+        - PyTree leaf types (such as torch.Tensor, int, float, bool)
+        - Data structures that can be flattened by PyTree
+            (e.g., dict, list, tuple, namedtuple, OrderedDict)
+        - Dataclass subclass (must be registered with 
+            torch._export.utils.register_dataclass_as_pytree_node` first)
+
+    torch.export() specializes the traced program based on the values of 
+    inputs that are not torch.Tensors. If you want to preserve dynamic
+    branching behavior in the generated graph, you will need to manually
+    rewrite the source code using control flow operations like
+    `torch.ops.higher_order.cond`.
+
+    While tracing, torch.export() takes note of shape-related assumptions 
+    made by the user program and the underlying PyTorch operator kernels.
+    The output ExportedProgram is considered valid only when these
+    assumptions hold true.
+
+    The returned ExportedProgram maintains the following invariants:
+        - It is guaranteed to be a faithful representation of the original
+            program.
+        - It maintains the exact calling convention of the original program.
+        - It contains a `state_dict` that stores the `torch.nn.Parameters`
+            of the original program.
+        - It includes an fx.GraphModule that represents the computation of
+            the original program. The GraphModule:
+            - Contains only `placeholder`, `call_function`, and `return` nodes.
+            - Inlines all submodules from the original programs.
+            - Lifts all parameters and buffers of the original program as
+                inputs to the graph.
+            - Does not mutate intermediate values, parameters, or buffers.
+            - Does not include operations with side effects.
+            - Contains only a curated subset of ATen operations and registered
+                custom operations (by default).
+                See the list of Core ATen Ops here: https://pytorch.org/docs/stable/ir.html
+
+    Args:
+        f: The `nn.Module` or callable to trace. If an `nn.Module` is provided,
+             its `forward` method will be traced.
```

Comment:
Technically, `__call__` is traced, no? Which if it has hooks, those will get traced, amirite?

Reply:
@voznesenskym that's not my recollection for forward module hooks?